test/e2e-preferred-host/ks_preferred_host_kas_test.go

```diff
@@ -1,7 +1,188 @@
 package e2e_preferred_host
 
-import "testing"
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"github.com/stretchr/testify/require"
+	"math/rand"
+	"net"
+	"net/url"
+	"strconv"
+	"testing"
+	"time"
+
+	v1 "k8s.io/api/core/v1"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/fields"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/apimachinery/pkg/util/wait"
+	"k8s.io/apimachinery/pkg/watch"
+	clientset "k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/tools/cache"
+	watchtools "k8s.io/client-go/tools/watch"
+
+	configv1client "github.com/openshift/client-go/config/clientset/versioned"
+	operatorv1client "github.com/openshift/client-go/operator/clientset/versioned/typed/operator/v1"
+	"github.com/openshift/cluster-kube-scheduler-operator/pkg/operator/operatorclient"
+	test "github.com/openshift/cluster-kube-scheduler-operator/test/library"
+	testlibrary "github.com/openshift/library-go/test/library"
+)
+
+var (
+	// the following parameters specify for how long apis must
+	// stay on the same revision to be considered stable
+	waitForKSORevisionSuccessThreshold = 3
+	waitForKSORevisionSuccessInterval  = 1 * time.Minute
+
+	// the following parameters specify max timeout after which
+	// KSOs are considered to not converged
+	waitForKSORevisionPollInterval = 30 * time.Second
```

Comment:
this is more like `KS`, we will be waiting for the scheduler pods, not the operator.